### DISCUSSION TITLE
Parameters in OpenAi like API

### DISCUSSION THREAD

**Discussion by GeroWeinkauf** [oobabooga/text-generation-webui#4946] [Q&A]
Hello im seeking for help in order to advance in my little project! Thanks in advance.

I have set up the WebUI on my local machine and was able to communicate with it via the open ai like API. (I have a small chatbot implementation running in js)

I#m currently wondering why the output differs that much between using this chatbot and a request with the chat in the WEBUI tab? Do I have to specify all the parameters in the API request in order to run the same parameters or are they global (for both "chats").

And if I have to specifiy them, do you have a json sample? 

Cheers

**Comment by mabrown666** [comment]
From my experience no, you do not have to specify then. But it doesn't hurt.
You can get a look at the parameters from the parameters tab, then hitting the save button as if you were creating a new parameter set (based off what is selected). You can then copy the parameter list and hit cancel.
Or if it's helpful you can use mine
{"max_tokens": 600, "stream": true, "do_sample": true, "temperature": 0.72, "top_p": 0.73, "typical_p": 1, "epsilon_cutoff": 0, "eta_cutoff": 0, "repetition_penalty": 1.1, "encoder_repetition_penalty": 1, "top_k": 0, "min_length": 0, "no_repeat_ngram_size": 0, "num_beams": 1, "penalty_alpha": 0, "length_penalty": 1, "early_stopping": false, "mirostat_mode": 0, "mirostat_tau": 5, "mirostat_eta": 0.1,
"tfs": 1, "top_a": 0}

But the most likely issue is probably based in the mode you are using. If you are using the chat interface, then you will need to manage the chat history for your API calls. If you are using the notebook interface (otherwise known as the instruct mode {"mode":"instruct"} ) then you should be able to simply copy your notebook text into the prompt element of your json and get the same response.

Hope that kind of helps, but I'm no expert in this field.
